bubblegum/program/README.md

```diff
@@ -1,4 +1,438 @@
+
+Compressed NFTs are secured on-chain by the hashing of the state data when it is added to the Merkle tree.  The Merkle root is a hash that cryptographically secures the state data for all of the leaves (NFTs) contained in the tree.
+
+In the unlikely scenario that all RPC providers were to lose their data stores, the off-chain state of compressed NFTs could be recovered by replaying transactions (provided that the given tree was started from the beginning).
+
+Compressed NFTs can also be losslessly decompressed into uncompressed Metaplex NFTs.  Decompression will cost rent for the token-metadata Metadata and Master Edition accounts that need to be created.
+
+## Basic operation
+
+### Creating and minting to a tree
+
+Anyone can create a tree using `create_tree` and then they are the tree owner.  They can also delegate authority to another wallet.
+
+### Merkle proofs
+
+After an NFT is minted, for any operations that modify the NFT, Merkle proofs must be provided with the instruction to validate the Merkle tree changes.  Bubblegum is an Anchor program and makes use of the remaining accounts feature for this purpose.  Merkle proofs are added as remaining account `Pubkey`s that are 32-byte Keccak256 hash values that represent the nodes from the Merkle tree that are required to calculate a new Merkle root.
+
+### Creator verification
+
+Creators are specified in a creators array in the metadata passed into the `mint_v1` instruction.  All creators for which `verified` is set to true in the creators array at the time of mint must be a signer of the transaction or the mint will fail.  Beyond the signers specified in the `MintV1` Account validation structure, the `mint_v1` instruction uses remaining accounts to optionally look for additional signing creators.  This does not conflict with the Merkle proofs requirement because proofs are not required for minting.
+
+Beyond verifying creators at the time of mint, there are `verify_creator` and `unverify_creator` instructions that can be used on existing Compressed NFTs.
+
+### Collection verification
+
+Note that there is no such thing as compressed Verified Collections.  Collections are still NFTs created in the realm of token-metadata Metadata and Master Edition accounts.  Also note that a collection cannot be set to verified at the time of minting.  Instead, there are instructions to `verify_collection` and `unverify_collection`, as well as a `set_and_verify_collection` instruction for the case where the collection was set during the mint.  All of these require either the true Collection Authority to be a a signer, or a delegated Collection Authority to be a signer along with providing a Collection Authority Record PDA.  See the Metaplex documentation on [`Certified Collections`](https://docs.metaplex.com/programs/token-metadata/certified-collections) for more information on verifying collections.
```

Comment:
> the realm of token-metadata Metadata and Master Edition accounts.

Same nit here with the double sequential "metadata" in the sentence.

Reply:
done